privatek8s.tf

```diff
@@ -18,6 +18,8 @@ data "azurerm_subnet" "privatek8s_tier" {
   name                 = "privatek8s-tier"
   resource_group_name  = data.azurerm_resource_group.private.name
   virtual_network_name = data.azurerm_virtual_network.private.name
+  # Enable KeyVault service endpoint so the cluster can access secrets to update other clusters
+  service_endpoints = ["Microsoft.KeyVault"]
```

Comment:
this is a data resource...? you can't do that here

Reply:
https://github.com/jenkins-infra/azure-net/pull/34